File: Sandbox.ApiService/CustomerModule/CustomerAddress.cs
namespace Sandbox.ApiService.CustomerModule;

public abstract class CustomerAddress
{
    public int Id { get; init; }
    public abstract string Type { get; }
    public Address Address { get; private set; }

    protected CustomerAddress(Address address)
    {
        Address = address;
    }
    protected CustomerAddress() { }
}

File: Sandbox.AngularWorkspace/projects/sandbox-app/src/app/customer-management/shared/customer-address/customer-address.ts
import { ChangeDetectionStrategy, Component, input } from '@angular/core';
import { type FieldTree, FormField } from '@angular/forms/signals';
import { formValidation } from '@form-validation';
import { type Address } from '@sandbox-app/customer-management/models';

@Component({
	selector: 'sandbox-customer-address',
	imports: [FormField, formValidation],
	templateUrl: './customer-address.html',
	changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CustomerAddress {
	public readonly address = input.required<FieldTree<Address>>();
}
